js/src/logger.ts

```diff
@@ -3743,34 +3744,43 @@ class ObjectFetcher<RecordType>
       const state = await this.getState();
       let data = null;
       if (this._internal_btql) {
-        const resp = await state.apiConn().post(
-          `btql`,
-          {
-            query: {
-              ...this._internal_btql,
-              select: [
-                {
-                  op: "star",
-                },
-              ],
-              from: {
-                op: "function",
-                name: {
-                  op: "ident",
-                  name: [this.objectType],
-                },
-                args: [
+        let cursor = undefined;
+        while (true) {
```

Comment:
may want to limit the # of iterations here (say to 10,000) and throw an exception after that.